packages/protocol/contracts/stability/BreakerBox.sol

```diff
@@ -0,0 +1,305 @@
+pragma solidity ^0.5.13;
+
+import { IBreakerBox } from "./interfaces/IBreakerBox.sol";
+import { IBreaker } from "./interfaces/IBreaker.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+
+import { AddressLinkedList, LinkedList } from "../common/linkedlists/AddressLinkedList.sol";
+import { UsingRegistry } from "../common/UsingRegistry.sol";
+import { Initializable } from "../common/Initializable.sol";
+import { Exchange } from "./Exchange.sol";
+
+/**
+ * @title   BreakerBox
+ * @notice  The BreakerBox checks the criteria defined in separate breaker contracts
+ *          to determine whether or not buying or selling should be allowed for a
+ *          specified exchange. The contract stores references to all breakers
+ *          that hold criteria to be checked, exchanges that
+ *          can make use of the BreakerBox & their current trading
+ */
+contract BreakerBox is IBreakerBox, Initializable, UsingRegistry {
+  using AddressLinkedList for LinkedList.List;
+
+  /* ==================== State Variables ==================== */
+
+  address[] public exchanges;
+  // Maps exchange address to its current trading mode info
+  mapping(address => TradingModeInfo) public exchangeTradingModes;
+  mapping(uint64 => address) public tradingModeBreaker;
+  mapping(address => uint64) public breakerTradingMode;
+  // Ordered list of breakers to be checked.
+  LinkedList.List private breakers;
+
+  modifier validateBreaker(address breaker, uint64 tradingMode) {
+    require(!isBreaker(breaker), "This breaker has already been added");
+    require(
+      tradingModeBreaker[tradingMode] == address(0),
+      "There is already a breaker added with the same trading mode"
+    );
+    require(tradingMode != 0, "The default trading mode can not have a breaker");
+    _;
+  }
+
+  /* ==================== Constructor ==================== */
+
+  /**
+   * @notice Sets initialized == true on implementation contracts
+   * @param test Set to true to skip implementation initialization
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @param breaker The address of a breaker to be added
```

Comment:
Agree with you on this. Initially had an array then thought it would be unnecessary code as we only have one breaker. Will remove from the initializer completely.